protocols/imap/src/main/java/org/apache/james/imap/processor/AppendProcessor.java

```diff
@@ -126,15 +126,14 @@ private void appendToMailbox(InputStream message, Date datetime, Flags flagsToBe
         try {
             final MailboxSession mailboxSession = session.getMailboxSession();
             final SelectedMailbox selectedMailbox = session.getSelected();
-            final MailboxManager mailboxManager = getMailboxManager();
             final boolean isSelectedMailbox = selectedMailbox != null && selectedMailbox.getMailboxId().equals(mailbox.getId());
             final ComposedMessageId messageId = mailbox.appendMessage(message, datetime, mailboxSession, !isSelectedMailbox, flagsToBeSet);
             if (isSelectedMailbox) {
                 selectedMailbox.addRecent(messageId.getUid());
             }
 
             // get folder UIDVALIDITY
-            UidValidity uidValidity = mailboxManager.getMailbox(mailboxPath, mailboxSession)
+            UidValidity uidValidity = mailbox
```

Comment:
you won't have the last value of uidValidity anymore. Can it be a problem?

Reply:
We actually now ensure we return the UID Validity of the mailbox we appended the message to, it's no longer subject to data races.

See below it's usage:

```
okComplete(request, ResponseCode.appendUid(uidValidity, new UidRange[] { new UidRange(messageId.getUid()) }), responder);
```

It's thus even "more correct"

Reply:
We actually now ensure we return the UID Validity of the mailbox we appended the message to, it's no longer subject to data races.

See below it's usage:

```
okComplete(request, ResponseCode.appendUid(uidValidity, new UidRange[] { new UidRange(messageId.getUid()) }), responder);
```

It's thus even "more correct"